mobile-widgets/qml/DownloadFromDiveComputer.qml

```diff
@@ -162,21 +162,47 @@ Kirigami.Page {
 					elide: Text.ElideRight
 				}
 				onCurrentTextChanged: {
+					var curVendor
+					var curProduct
+					var curDevice
 					dc1.enabled = dc2.enabled = dc3.enabled = dc4.enabled = true
 					for (var i = 1; i < 5; i++) {
+						switch (i) {
+							case 1:
+								curVendor = PrefDiveComputer.vendor1
+								curProduct = PrefDiveComputer.product1
+								curDevice = PrefDiveComputer.device1
+								break
+							case 2:
+								curVendor = PrefDiveComputer.vendor2
+								curProduct = PrefDiveComputer.product2
+								curDevice = PrefDiveComputer.device2
+								break
+							case 3:
+								curVendor = PrefDiveComputer.vendor3
+								curProduct = PrefDiveComputer.product3
+								curDevice = PrefDiveComputer.device3
+								break
+							case 4:
+								curVendor = PrefDiveComputer.vendor4
+								curProduct = PrefDiveComputer.product4
+								curDevice = PrefDiveComputer.device4
+								break
+						}
+
```

Comment:
Having read especially one of your comments (in another PR), I see the need to explain this change. I could have choosen to make "yet another round of ......", by making a changing vendor etc in qPrefDiveComputer to a qStringList, which would have been the pure solution,  but I choose NOT to do that and instead made a more ugly implementation.

I hope this demonstrates, that I do NOT make changes for reason of purity or just because I have a firm opinion, I make changes because it moves subsurface-mobile forward in a positive manner.